crates/client/db/src/lib.rs

```diff
@@ -230,15 +349,15 @@ impl DeoxysBackend {
         BACKEND_SINGLETON.get().map(|backend| &backend.sierra_classes).expect("Backend not initialized")
     }
 
-    pub fn bonsai_contract() -> &'static Arc<Mutex<BonsaiStorage<BasicId, BonsaiDb, Pedersen>>> {
+    pub fn bonsai_contract() -> &'static BonsaiStorageAccess<Pedersen> {
```

Comment:
Once we've merged this I'll start working on integrating this into #28: do you think it would be a good idea to delegate mutability control into the `StorageHandler` struct (see #28)? ie:

- `StorageHandler` would contain a single static readable reference for each bonsai trie.
- When mutable access is required, a clone of the backing bonsai trie woudl be created.
- Merging back into the backing trie would be done by `StorageHandler`. This can be done in a safer way by giving a unique type to writable tries so that class trie updates cannot be merged back to the contract trie for example.

What is your opinion on this?

Reply:
If i understand correctly, `BonsaiStorageAccess` is already pretty much your `StorageHandler` in spirit without all the trie-specific utility functions, is that correct?